app/javascript/packs/proCharts.js

```diff
@@ -1,101 +1,125 @@
 import Chart from 'chart.js';
+import { Object } from 'es6-shim';
 
 const reactionsCanvas = document.getElementById('reactionsChart');
 const commentsCanvas = document.getElementById('commentsChart');
 
-export const reactionsChart = new Chart(reactionsCanvas, {
-  type: 'line',
-  data: {
-    labels: JSON.parse(reactionsCanvas.dataset.labels),
-    datasets: [
-      {
-        label: 'Total',
-        data: JSON.parse(reactionsCanvas.dataset.totalCount),
-        // data: [5, 10, 15, 17, 25, 23],
-        fill: false,
-        borderColor: 'rgb(75, 192, 192)',
-        lineTension: 0.1,
-      },
-      {
-        label: 'Likes',
-        data: JSON.parse(reactionsCanvas.dataset.totalLikes),
-        // data: [2, 5, 10, 10, 15, 13],
-        fill: false,
-        borderColor: 'rgb(229, 100, 100)',
-        lineTension: 0.1,
-      },
-      {
-        label: 'Unicorns',
-        data: JSON.parse(reactionsCanvas.dataset.totalUnicorns),
-        // data: [1, 2, 2, 4, 5, 3],
-        fill: false,
-        borderColor: 'rgb(157, 57, 233)',
-        lineTension: 0.1,
-      },
-      {
-        label: 'Bookmarks',
-        data: JSON.parse(reactionsCanvas.dataset.totalReadinglist),
-        // data: [2, 3, 3, 3, 5, 7],
-        fill: false,
-        borderColor: 'rgb(10, 133, 255)',
-        lineTension: 0.1,
-      },
-    ],
-  },
-  options: {
-    legend: {
-      position: 'bottom',
-    },
-    responsive: true,
-    title: {
-      display: true,
-      text: 'Reactions this Week',
-    },
-    scales: {
-      yAxes: [
-        {
-          ticks: {
-            suggestedMin: 0,
-            precision: 0,
+const padDate = month => String(`00${month}`).slice(-2);
+
+const oneWeekAgo = new Date();
+oneWeekAgo.setDate(oneWeekAgo.getDate() - 7);
+
+console.log(oneWeekAgo)
+
+fetch(
+  `/api/analytics/historical?start=${oneWeekAgo.getFullYear()}-${padDate(
```

Comment:
I wonder if this could be simplified by using `oneWeekAgo.toISOString().split("T")[0]`

Reply:
We're going to move over to some sort of JS time library anyways I think -- we're going to make different dates available and the math gets gnarly pretty fast!

Reply:
Consider looking into luxon (newer/better moment.js) or date-fns. Immutable dates plus some other nice features.

Reply:
+1 for date fns which can be tree shaken to include only single used functions. I heard about dayjs as well but I've never used it.

This is a great resource to compare libraries, their sizes and features: https://github.com/you-dont-need/You-Dont-Need-Momentjs/blob/master/README.md